### ISSUE TITLE
Error downloading the subtitle on Google TV & Mac

### ISSUE THREAD

**Issue by rickeylohia** [opensubtitlesdev/service.subtitles.opensubtitles-com#6]
I get below error when trying to download the subtitle after search on Google TV & Mac. But it works fine on FireTV

`2023-05-27 17:37:28.876 T:1807095   error <general>: OnDownloadComplete - Saving of subtitle /Users/<username>/Library/Application Support/Kodi/userdata/addon_data/service.subtitles.opensubtitles-com/temp/a8c8ea23-c0fc-4193-861b-3a8158b550c0.srt to special://temp/playlist.en.srt failed`

<img width="937" alt="Screenshot 2023-05-27 at 5 37 27 PM" src="https://github.com/opensubtitlesdev/service.subtitles.opensubtitles-com/assets/22551650/6893b9bb-a37a-46ab-aaad-b358cb632829">

**Comment by opensubtitlesdev** [comment]
-- reply above this line --
DEAR OPENSUBTITLESDEV/SERVICE OPENSUBTITLES-COM,
────────────────────────────────────────────────
Probably storage is full or doesnt gave access there



Your OpenSubtitles Team,
Please support us and [B̲e̲c̲o̲m̲e̲ V̲I̲P̲ m̲e̲m̲b̲e̲r̲](https://www.opensubtitles.org/en/support#vip)
[w̲w̲w̲.o̲pe̲n̲s̲u̲b̲t̲i̲t̲l̲e̲s̲.c̲o̲m̲](http://www.opensubtitles.com/) | [w̲w̲w̲.o̲pe̲n̲s̲u̲b̲t̲i̲t̲l̲e̲s̲.o̲r̲g]
(https://www.opensubtitles.org)
[A̲P̲I̲ f̲o̲r̲ d̲e̲v̲e̲l̲o̲pe̲r̲s̲](https://api.opensubtitles.com/)
──────────────────────────────────────────────────────────────────────────────────────────
We hope this response has sufficiently answered your questions. If not, please do not send
 another email. Instead, reply to this email or [l̲o̲gi̲n̲ t̲o̲ yo̲u̲r̲ a̲c̲c̲o̲u̲n̲t̲][1] for a complete 
                   archive of all your support requests and responses.

---------------------------------------------
[1] http://support.opensubtitles.org/view.php?auth=o1xdbiqaadetuaaalpLVRu6sZBMPmg%3D%3D
Ref-Mid: ***@***.***

**Comment by rickeylohia** [comment]
Does not seem to be a storage or permission issue since its working for addon `Opensubtitle.org`

**Comment by Seroczynski** [comment]
I am having the same issue on my OSMC Vero 4k+ device with the latest version of both Kodi and the plugin.

**Comment by tsboj17** [comment]
Same here and here is my setup:
Opensubtitles.com plugin, 1.0.1
Kodi 20.2.0 with OSMC
Raspberry 3 B

**Comment by CakeSpork** [comment]
I'm still having this problem on 1.0.2 at the moment. Haven't been able to use the opensubtitles.com version successfully yet at all. My logs are basically identical to the original post, and the error message is also the same.

**Comment by opensubtitlesdev** [comment]
are you able to download subtitles with another extensions, for example opensubtitles.org ?

this error message indicates the file can't be saved locally

**Comment by rickeylohia** [comment]
I think this is a not an issue with the add on but user related.
at least for me the credentials were incorrect. it seems the old add on `opensubtitles.org` accepts both `username/email` in the `username` field but the new one only accepts `username`.
And it seems the search uses hard coded `ApiKey` and not the credentials.  Only when downloading it uses the credentials, giving a false sense that credentials work. 
The error message can be better though on the download, it should say `Incorrect Credential` rather than `Error searching for subtitles`

**Comment by opensubtitlesdev** [comment]
> I think this is a not an issue with the add on but user related. at least for me the credentials were incorrect. it seems the old add on `opensubtitles.org` accepts both `username/email` in the `username` field but the new one only accepts `username`. 

Yes it is the case. we had to disallow the emails to improve performances and prevent spamming issues.  We clearly indicated it in the front screen of the application, and wrote an article to explain how to setup the plugin
https://blog.opensubtitles.com/opensubtitles/how-to-install-the-opensubtitles-kodi-add-ons 

> And it seems the search uses hard coded `ApiKey` and not the credentials. Only when downloading it uses the credentials, giving a false sense that credentials work. 

Yes, the search allows downloading with the api key only, this is to allow anonymous downloads. If a user doesn't enter credentials, he'll get 5 downloads before being prompted to register

> The error message can be better though on the download, it should say `Incorrect Credential` rather than `Error searching for subtitles`

Yeah, for that the messages could be better. The messages from the api could actually allow a better integration, but we are stucked on the python/kodi part to make a better implementation.

I just tried, and agree it's confusing, problem is that as it is possible to download without authentication, kodi keeps on trying downloading with invalid credentials, at this stage I can only catch the 406 error for download limit reached, and not the 401 from the login failed (actually, to my understanding in the code, it should catch it, but somehow this only works if I disallow anonymous downloads for all kodi users on the API side).  That could get us a lot of registrations, but we still think it's nicer to let users enjoy a few subtitles without requiring anything from them, there was plenty of complaints when we forced to login to use the old extension.
